NickvisionMoney.GNOME/Views/TransferDialog.cs

```diff
@@ -278,4 +287,28 @@ private bool OnKeyPressedDest(Gtk.EventControllerKey sender, Gtk.EventController
         }
         return false;
     }
+
+
+    private void PopulateRateFields()
+    {
+        if (_conversionRateGroup.Visible)
+        {
+            var rates = _controller.GetConversionRatesAsync().Result;
+            if (rates != null)
+            {
+                _sourceCurrencyRow.SetText(
+                    rates.Value.Item1.ToAmountString(_controller.CultureForSourceNumberString, 
+                        _controller.UseNativeDigits, 
+                        showCurrencySymbol: false));
+                
+                _controller.Transfer.ConversionRate = 1 / rates.Value.Item2;
+
+                var text = rates.Value.Item2.ToAmountString(_controller.CultureForSourceNumberString, 
+                    _controller.UseNativeDigits, 
+                    showCurrencySymbol: false);
+                
+                _destinationCurrencyRow.SetText(text);
```

Comment:
`GLib.Functions.IdleAdd(0, () => _destinationCurrencyRow.SetText(text))`, so it will run on main thread, then you can safely make this method async.

Reply:
Ha, thanks. Maybe thats why async wasn't working well for me.